plugin/graph_processing/README.md

```diff
@@ -0,0 +1,22 @@
+# Graph processing
+
+These plugins create and process cell graphs to make downstream predictions.
+
+## Development process
+
+Addition of new graph processing plugins to SPT is done in three steps:
+1. **Exploratory stage**: a user evaluates their processing model by manually using the output and upload functions described in the following section and examining their results locally or in an SPT instance after upload.
+2. **Proposal stage**: If a user determines that their graph model is a good candidate for inclusion into SPT as a default graph deep learning plugin, that could be run on every new study imported into SPT, they would containerize their model according to the Docker template defined in `template/`, upload it to a fork of this repository, and open an issue or pull request for SPT to include their plugin.
+3. **Inclusion stage**: The SPT maintainers would review the proposed plugin and, if it is accepted, upload the container to the SPT Docker page and modify SPT code to pull down and use that Docker image by default in SPT.
+
+## Graph processing plugins should look like this
+
+Graph processing plugins are to be made available as Docker images, built from a Dockerfile following the template provided in `Dockerfile`.
+
+Each plugin should have the following commands available from anywhere in the Docker image:
+* `spt-plugin-print-graph-request-configuration`, which prints to `stdout` the configuration file intended to be used by this plugin to fetch graphs from an SPT instance to use for model training. An empty configuration file and a shell script to do this is provided in this repo, as well as the command needed to make this available in the template `Dockerfile`.
```

Comment:
It looks like the Dockerfile uses an ADD instruction to place `graph.config` into the Docker image of the built plugin, e.g. the `cg-gnn` plugin. But this file is only a template (empty `target_name`). How are the contents of `graph.config` to be specified at runtime for different datasets using the same Docker image?

Reply:
I forgot to update it when it says "this repo" to "the template" since this was originally the README for this template only. The `graph.config` file is not used within the Docker images, it's a (suggested default) setting file for the user to use with SPT when extracting the graph files to run the image(s) on.

Reply:
Ah, yes that is what I thought.
But then why is `spt-plugin-print-graph-request-configuration` used?

Reply:
It's so you can get a `graph.config` file to use. It prints to stdout and then you copy it into your local directory and run `spt graph extract` with it.

Reply:
But the file it emits is never directly useable, right? If the idea is to manually edit the emitted template just before doing some run, it seems simpler to use the source code version of `graph.config` directly from this repository. The main runtime usage of the Docker container, to do its calculation, doesn't use the `spt-plugin-print-graph-request-configuration` command, if I understand correctly.

I think a similar remark applies to `spt-plugin-print-training-configuration`. What is the rationale for incorporating some of these development-specific actions into the docker image? Is it to avoid the need to access git source code in addition to a Docker image pull? If this is the case, we can perhaps make these 2:

- `spt-plugin-print-graph-request-configuration`
- `spt-plugin-print-training-configuration`

into commands provided by the template image only, and not indicated as "required" by the plugins themselves (since they are not required for the main action, training/processing).

Reply:
I put them in at your request a while back. I believe the rationale was to provide a way to get the recommended default settings without having to muck about with the Docker internals / source code.

The training variables `cg-gnn` wants are different from the training variables `graph-transformer` wants, and likewise the graph formats plugins want may be different from plugin to plugin (although in practice they're identical so far), so making it a template command would not work.

Reply:
I see, yeah this seems like the best that can be done now.